server/channels/store/sqlstore/integrity.go

```diff
@@ -544,3 +547,61 @@ func CheckRelationalIntegrity(ss *SqlStore, results chan<- model.IntegrityCheckR
 	mlog.Info("Done with relational integrity checks")
 	close(results)
 }
+
+func GetAllValidUserIDsFromDMChannels(ss *SqlStore) ([]*model.User, error) {
+	records := []model.OrphanedRecord{}
+	dirtyIDs := []string{}
+	query, args, err := ss.getQueryBuilder().
+		Select().
+		Column("CT.ChannelName AS ParentId").
+		From("Channels AS CT").
+		Where(sq.Eq{"CT.Type": []model.ChannelType{model.ChannelTypeDirect}}).
+		ToSql()
+
+	if err != nil {
+		mlog.Error("There is an error with the DM Channel Name query")
+		return nil, err
+	}
+
+	err = ss.GetMasterX().Select(&records, query, args...)
+	if err != nil {
+		mlog.Error("there us an issue fetching the records for DM Channel Names")
+		return nil, err
+	}
+
+	for k := range records {
+		dirtyIDs = append(dirtyIDs, *records[k].ParentId)
+	}
+
+	userIDs := []string{}
+
+	for _, v := range dirtyIDs {
+		temp := strings.Split(v, "_")
```

Comment:
You have to split at `__`, not `_`.

Reply:
oof my bad, I just assummed there was only one underscore. 😞